builtin/settingtypes.txt

```diff
@@ -819,6 +819,9 @@ server_url (Server URL) string https://minetest.net
 #    Automatically report to the serverlist.
 server_announce (Announce server) bool false
 
+#    Hide player names from the serverlist by not sending them.
+server_announce_anonymize_names (Hide player names on the serverlist) bool false
```

Comment:
Positive settings are easier to understand. So I recommend

`server_announce_send_players (Send player names to the server list)`

Reply:
personally I think it's fine.
this isn't something confusing like `server_accounce_dont_show_players`

Reply:
@rubenwardy If you wish, you can merge [`20240706-server-anon-name!positive_settings`](https://github.com/Emojigit/minetest/tree/20240706-server-anon-name!positive_settings) into this PR. I am OK with both approaches.

Reply:
Seems like the positive settings proposal is accepted by more devs. I will apply the changes soon.

Reply:
Done in https://github.com/Emojigit/minetest/commit/2f03ec681c4e22f1555bde41cc2e4cc50fab7bf5